components/leetcode-area.tsx

```diff
@@ -59,10 +68,19 @@ const LeetCodeArea = () => {
 
     useEffect(() => {
         if (!leetcodeUserName) return;
-        console.log('leetcodeUserName ', leetcodeUserName);
         getLeetCodeInfo();
     }, [leetcodeUserName]);
 
+    useEffect(() => {
+        if (userFound.leetcode_url) return;
```

Comment:
@JRakin before putting data need to check wether its new or old data. if the data is new ( newly fetched from leetcode api) only then we need to save to reduce unnecessary call to our database

Reply:
okay @sifatul vaiya I'll do that

Reply:
@sifatul we not calling leetcode api when we have data in DB by that username

Reply:
@JRakin please check again. i think it will store data twice.
For example, your a userA is in database . 
1st render: this useEffect will be called and returned as redux does not have the data yet
2nd render: this use effect will be called again as we got data form database and updated redux:  leetcodeusername and profile url got  updated. This time it can pass the condition or not?